lazy-pages/src/lib.rs

```diff
@@ -20,68 +20,142 @@
 //! In runtime data for contract wasm memory pages can be loaded in lazy manner.
 //! All pages, which is supposed to be lazy, must be mprotected before contract execution.
 //! During execution data from storage is loaded for all pages, which has been accesed.
-//! See also `handle_sigsegv`.
+//! See also `user_signal_handler`.
+//!
+//! Restrict any page handling in signal handler more then one time.
+//! If some page will be released twice it means, that this page has been added
+//! to lazy pages more then one time during current execution.
+//! This situation may cause problems with memory data update in storage.
+//! For example: one page has no data in storage, but allocated for current program.
+//! Let's make some action for it:
+//! 1) Change data in page: Default data  ->  Data1
+//! 2) Free page
+//! 3) Alloc page, data will Data2 (may be equal Data1).
+//! 4) After alloc we can set page as lazy, to identify wether page is changed after allocation.
+//! This means that we can skip page update in storage in case it wasnt changed after allocation.
+//! 5) Write some data in page but do not change it Data2 -> Data2.
+//! During this step signal handler writes Data2 as data for released page.
+//! 6) After execution we will have Data2 in page. And Data2 in released. So, nothing will be updated
+//! in storage. But program may have some significant data for next execution - so we have a bug.
+//! To avoid this we restrict double releasing.
+//! You can also check another cases in test: memory_access_cases.
+
+#![allow(useless_deprecated, deprecated)]
```

Comment:
> useless_deprecated

Why?

Reply:
We mark code as deprecated, but use it because of old runtimes support

Reply:
This is what `deprecated` for but I'm talking about `useless_deprecated` which ocurrs when `#[deprecated]` has no effect

Reply:
> This is what `deprecated` for but I'm talking about `useless_deprecated` which ocurrs when `#[deprecated]` has no effect

Runtime interface macro calls all fns under the hood, so it fails for clippy as usage of deprecated fns.
In @ukint-vs I offered to use this allowance for whole runtime-interface crate: that we could mark something as deprecated and still use it.